### PR TITLE
Merging aggregator back to master

### PR DESCRIPTION
**Pull request by til0n**

### CODE REVIEW

File: src/main/java/org/moonlightcontroller/aggregator/ApplicationAggregator.java [hunk 1]
```diff
@@ -1,146 +1,718 @@
...
+	@Override
+	public void performAggregation() {
+		synchronized (this) {
+			// TODO: The following should be done for each OBI location specifier
+			List<InstanceLocationSpecifier> obiLocationSpecifiers = TopologyManager.getInstance().getAllEndpoints(); 
+			for (InstanceLocationSpecifier loc : obiLocationSpecifiers) {
+				IProcessingGraph last = null;
+				for (BoxApplication app : this.apps) {
+					// TODO: Complete here
+					// current = app.getProcessingGraph(); // This method does not exist now
```
Comment (yotamhc) [inline src/main/java/org/moonlightcontroller/aggregator/ApplicationAggregator.java:706]: We should complete this. Please let me know if you need me for that.

File: src/main/java/org/moonlightcontroller/blocks/Alert.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			Alert o = (Alert)other;
+			return new Alert("MERGED##" + this.getId() + "##" + other.getId() + "##" + UUIDGenerator.getSystemInstance().getUUID().toString(), this.message + ";;" + o.message);
+		} else {
+			throw new MergeException("Cannot merge statics of different type");
+		}
+	}
+	
+	@Override
+	protected void putConfiguration(Map<String, String> config) {
+		// TODO: Does alert have config?
```
Comment (yotamhc) [inline src/main/java/org/moonlightcontroller/blocks/Alert.java:58]: Yes, see part 5.16 in the OpenBox spec (we can ignore optional parameters)

File: src/main/java/org/moonlightcontroller/aggregator/ApplicationAggregator.java (hunk 1 above)
Comment (til0n) [inline src/main/java/org/moonlightcontroller/aggregator/ApplicationAggregator.java:706]: Not sure what should be done here.
Why should it be done for each OBI.
I think it's better to not block this merge on it.
I will do it in my next merge.

Comment (yotamhc) [inline src/main/java/org/moonlightcontroller/aggregator/ApplicationAggregator.java:706]: I am merging the PR anyway. Don't worry.

**Comment by yotamhc** [comment]
Good job!

**Pull request merged**